<launch>
  <arg name="move_forward_only" default="false"/>
  <arg name="cmd_vel_topic" default="cmd_vel" />
  <arg name="odom_frame_id" default="odom" />
  <arg name="base_frame_id" default="base_footprint" />
  <arg name="laser_topic" default="scan" />
  <arg name="odom_topic" default="odom" />
  <arg name="global_frame_id" default="map" />
  
  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_global_planner" value="navfn/NavfnROS" />
    <!--param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/-->
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find minhtien)/param/costmap_common_params_burger.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find minhtien)/param/costmap_common_params_burger.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find minhtien)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find minhtien)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find minhtien)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find minhtien)/param/dwa_local_planner_params_waffle.yaml" command="load" />
 
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)" />
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)" />
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)" />
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)" />
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)" />
    
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    
  </node>
</launch>
